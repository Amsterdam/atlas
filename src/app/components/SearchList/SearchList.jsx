import React from 'react'
import PropTypes from 'prop-types'
import { toDetailFromEndpoint } from '../../../store/redux-first-router/actions'
import SearchListItem from '../SearchListItem/SearchListItem'
import { VIEW_MODE } from '../../../shared/ducks/ui/ui'

const SearchList = ({ categoryResults, limit }) => {
  const results =
    categoryResults && categoryResults.results
      ? categoryResults.results.map((result) => ({
          ...result,
          linkTo: toDetailFromEndpoint(result.endpoint, VIEW_MODE.SPLIT),
        }))
      : []

  return (
    <div className="qa-search-results-list">
      <ul className="o-list">
        {results.slice(0, limit).map((result, i) => (
          <SearchListItem
            key={`${result.label}-${i}`} // eslint-disable-line react/no-array-index-key
            {...{
              category: categoryResults,
              result,
            }}
          />
        ))}
      </ul>
    </div>
  )
}

SearchList.propTypes = {
  categoryResults: PropTypes.shape({}).isRequired,
  userScopes: PropTypes.arrayOf(PropTypes.string).isRequired,
  limit: PropTypes.number.isRequired,
}

export default SearchList
