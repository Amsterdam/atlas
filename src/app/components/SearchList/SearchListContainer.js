import { connect } from 'react-redux';
import {
  getSearchCategory,
  getSearchQueryResults
} from '../../../shared/ducks/data-search/selectors';
import SearchList from './SearchList';
import { fetchMoreResults } from '../../../shared/ducks/data-search/actions';

const mapStateToProps = (state) => ({
  category: getSearchCategory(state),
  searchResults: getSearchQueryResults(state)
});

const mapDispatchToProps = (dispatch) => ({
  fetchMoreResults: () => dispatch(fetchMoreResults())
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchList);
