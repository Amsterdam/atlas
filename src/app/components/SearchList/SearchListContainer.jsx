import { connect } from 'react-redux'
import SearchList from './SearchList'
import { getUserScopes } from '../../../shared/ducks/user/user'

const mapStateToProps = (state) => ({
  userScopes: getUserScopes(state),
})

export default connect(mapStateToProps)(SearchList)
