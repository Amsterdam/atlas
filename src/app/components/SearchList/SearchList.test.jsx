import React from 'react';
import { shallow } from 'enzyme';
import SearchList from './SearchList';

describe('SearchList', () => {
  it('should render a results list', () => {
    const props = {
      categoryResults: {
        results: [
          {
            label: 'Dam 1',
            endpoint: 'https://acc.api.data.amsterdam.nl/bag/verblijfsobject/03630003761571/'
          },
          {
            label: 'Dam 2',
            endpoint: 'https://acc.api.data.amsterdam.nl/bag/verblijfsobject/03630011909911/'
          }
        ]
      },
      fetchMoreResults: jest.fn(),
      hasLoadMore: null,
      limit: 10
    };

    const component = shallow(
      <SearchList {...props} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render an empty list ', () => {
    const props = {
      categoryResults: { },
      fetchMoreResults: jest.fn(),
      hasLoadMore: null,
      limit: 10
    };

    const component = shallow(
      <SearchList {...props} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the Toon meer button when the hasLoadMore flag is set ', () => {
    const props = {
      categoryResults: { },
      fetchMoreResults: jest.fn(),
      hasLoadMore: true,
      limit: 10
    };

    const component = shallow(
      <SearchList {...props} />
    );
    expect(component).toMatchSnapshot();
  });
});
