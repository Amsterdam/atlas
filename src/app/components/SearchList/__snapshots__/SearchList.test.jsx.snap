// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchList should render a results list 1`] = `
<div
  className="qa-search-results-list"
>
  <ul
    className="o-list"
  >
    <SearchListItem
      category={
        Object {
          "results": Array [
            Object {
              "endpoint": "https://acc.api.data.amsterdam.nl/bag/verblijfsobject/03630003761571/",
              "label": "Dam 1",
            },
            Object {
              "endpoint": "https://acc.api.data.amsterdam.nl/bag/verblijfsobject/03630011909911/",
              "label": "Dam 2",
            },
          ],
        }
      }
      key="Dam 1-0"
      result={
        Object {
          "endpoint": "https://acc.api.data.amsterdam.nl/bag/verblijfsobject/03630003761571/",
          "label": "Dam 1",
          "linkTo": Object {
            "meta": Object {
              "additionalParams": Object {
                "view": undefined,
              },
              "preserve": true,
            },
            "payload": Object {
              "id": "id03630003761571",
              "subtype": "verblijfsobject",
              "type": "bag",
            },
            "type": "atlasRouter/DATA_DETAIL",
          },
        }
      }
    />
    <SearchListItem
      category={
        Object {
          "results": Array [
            Object {
              "endpoint": "https://acc.api.data.amsterdam.nl/bag/verblijfsobject/03630003761571/",
              "label": "Dam 1",
            },
            Object {
              "endpoint": "https://acc.api.data.amsterdam.nl/bag/verblijfsobject/03630011909911/",
              "label": "Dam 2",
            },
          ],
        }
      }
      key="Dam 2-1"
      result={
        Object {
          "endpoint": "https://acc.api.data.amsterdam.nl/bag/verblijfsobject/03630011909911/",
          "label": "Dam 2",
          "linkTo": Object {
            "meta": Object {
              "additionalParams": Object {
                "view": undefined,
              },
              "preserve": true,
            },
            "payload": Object {
              "id": "id03630011909911",
              "subtype": "verblijfsobject",
              "type": "bag",
            },
            "type": "atlasRouter/DATA_DETAIL",
          },
        }
      }
    />
  </ul>
</div>
`;

exports[`SearchList should render an empty list  1`] = `
<div
  className="qa-search-results-list"
>
  <ul
    className="o-list"
  />
</div>
`;

exports[`SearchList should render the Toon meer button when the hasLoadMore flag is set  1`] = `
<div
  className="qa-search-results-list"
>
  <ul
    className="o-list"
  />
  <button
    className="c-show-more c-show-more--gray qa-show-more"
    onClick={[MockFunction]}
    tabIndex="0"
    type="button"
  >
    Toon meer
  </button>
</div>
`;
